cli/internal/taskhash/taskhash.go

```diff
@@ -272,12 +248,9 @@ func (th *Tracker) calculateDependencyHashes(dependencySet dag.Set) ([]string, e
 // that it has previously been called on its task-graph dependencies. File hashes must be calculated
 // first.
 func (th *Tracker) CalculateTaskHash(packageTask *nodes.PackageTask, dependencySet dag.Set, logger hclog.Logger, args []string, useOldTaskHashable bool) (string, error) {
-	pfs := specFromPackageTask(packageTask)
-	pkgFileHashKey := pfs.ToKey()
-
-	hashOfFiles, ok := th.packageInputsHashes[pkgFileHashKey]
+	hashOfFiles, ok := th.packageInputsHashes[packageTask.TaskID]
 	if !ok {
-		return "", fmt.Errorf("cannot find package-file hash for %v", pkgFileHashKey)
+		return "", fmt.Errorf("cannot find package-file hash for %v", packageTask.TaskID)
```

Comment:
This error message is now infinitely better.

Reply:
I'm surprised you didn't say "rounds-to-infinity" in true nathan style 😉 😛